cdi-core/src/main/java/com/linkedin/cdi/connection/S3Connection.java

```diff
@@ -130,32 +130,44 @@ public WorkUnitStatus execute(WorkUnitStatus status) {
     return status;
   }
 
-  private ByteArrayInputStream handleUpload(String finalPrefix) {
+  /**
+   * Get s3 key either from activation parameters named `s3key` or from the source path itself
+   */
+  @NotNull
+  private String getS3Key(Path path) {
```

Comment:
This is the main logic on how we get s3key from the secondary input. The config will be:

`[{"path": "${job.dir}/${preceding.table.name}", "fields":["pathName", "s3key"], "category": "activation"},{"path": "{{pathName}}", "category": "payload", "format":"binary"}]
`

Reply:
There may be better way to do this like adding another similar "binary" payload for file name, or make the binary payload a json object with path and name. But it won't be backward compatible. I also try to keep all the parameters for upload (which is a special case in the code) close together. Currently, the "binary" payload is only relevant while uploading.